utils/service_discovery/abstract_config_store.py

```diff
@@ -113,45 +117,65 @@ def get_check_tpls(self, image, **kwargs):
                 templates.append((c_name, init_config_tpls[idx], instance_tpls[idx]))
         return templates
 
-    def read_config_from_store(self, image):
+    def read_config_from_store(self, identifier):
         """Try to read from the config store, falls back to auto-config in case of failure."""
         try:
-            check_names = json.loads(
-                self.client_read(path.join(self.sd_template_dir, image, 'check_names').lstrip('/')))
-            init_config_tpls = json.loads(
-                self.client_read(path.join(self.sd_template_dir, image, 'init_configs').lstrip('/')))
-            instance_tpls = json.loads(
-                self.client_read(path.join(self.sd_template_dir, image, 'instances').lstrip('/')))
-            source = CONFIG_FROM_TEMPLATE
-        except (KeyNotFound, TimeoutError, json.JSONDecodeError) as ex:
-            # first case is kind of expected, it means that no template was provided for this container
-            if isinstance(ex, KeyNotFound):
+            res = self._issue_read(identifier)
+            if res and len(res) == 3:
+                source = CONFIG_FROM_TEMPLATE
+                check_names, init_config_tpls, instance_tpls = res
+            else:
+                log.debug("Could not find directory {0} in the config store, "
```

Comment:
small notice: since we dropped 2.6 support, we can avoid using indexes like `{0}` when not needed, `{}` is fine.